alsoHeresTheInterviewQuestionIHad/alsoHeresTheInterviewQuestionIHad.js

```diff
@@ -0,0 +1,31 @@
+const fs = require("fs");
+const streetArray = fs.readFileSync("data.txt", "utf-8").split(",");
+let answer = "";
+let tallestBuilding = Math.max(...streetArray);
+let numberOfBuildings = streetArray.length;
+let tallestBuildingSoFar = 0;
+let bestArea = tallestBuilding * numberOfBuildings;
+let bannerOne = [tallestBuilding, numberOfBuildings];
+let bannerTwo = [tallestBuilding, numberOfBuildings];
+let bestBannerOne = [tallestBuilding, numberOfBuildings];
+let bestBannerTwo = [tallestBuilding, numberOfBuildings];
+for (let i = 0; i < numberOfBuildings - 1; i++){
+    if (streetArray[i] > tallestBuildingSoFar) {
+        tallestBuildingSoFar = streetArray[i];
+    }
+    bannerOne = [tallestBuildingSoFar, i + 1];
+    bannerTwo = [Math.max(...streetArray.slice(i + 1, streetArray.length)), numberOfBuildings - i - 1];
+
+//    console.log("At address " + i + " the two banners required would be " + bannerOne + " and " + bannerTwo);
+
+    if (((bannerOne[0] * bannerOne[1]) + (bannerTwo[0] * bannerTwo[1])) < bestArea) {

```

Comment:
nit: the extra parens actually aren't needed here, but if you find it more readable 🤷

Reply:
Will I look old or feeble by having the extra brackets?  I do find it more readable and easier to follow the math blocks that way, but I could go through and deprecate them from code in future.  I actually even looked up if I could drop brackets from simple if statements altogether.  Spoiler: not in JS apparently, but it's okay in Python I think Themp said.

Reply:
It's personal preference

Reply:
If you install prettier and have it run on save you it will take out extra brackets that aren't needed

https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode

It stops most style related comments by automating it